src/main/java/login/LoginInteractor.java

```diff
@@ -1,27 +1,35 @@
 package login;
 
+import entity.Curr;
+import presenter.LoginPresenter;
 import entity.User;
-import presenter.LoginResponsePresenter;
 
 
-public class LoginInteractor {
+public class LoginInteractor<LoginReponseModel> implements LoginInputBoundary{
 
     final LoginDSGateway loginDsGateway;
-    final LoginResponsePresenter userPresenter;
+    final LoginPresenter userPresenter;
+    final Curr currentUser;
 
-    public static void login(LoginRequestModel logReqMod){
-        // if not gateway.userExists(logreqMod){
-        // LoginFailureResponseModel(
-    }
-
-    public LoginInteractor(LoginDSGateway loginDsGateway, LoginResponsePresenter userPresenter){
+    public LoginInteractor(LoginDSGateway loginDsGateway, LoginPresenter userPresenter, Curr currentUser){
         this.loginDsGateway = loginDsGateway;
         this.userPresenter = userPresenter;
+        this.currentUser = currentUser;
     }
 
-    public static User getUser(LoginRequestModel logReqMod) {
-        // User user = gateway.getUser(LoginRequestModel)
-        // LoginResponsePresenter.displayUser(User)
-        return null;
+    public void login(LoginRequestModel logReqMod) {
+        if (loginDsGateway.userExists(logReqMod)){
+            User user = loginDsGateway.getUser(logReqMod);
```

Comment:
When people login, we need setup the Curr.User
To achieve this, just add a like Curr.setUser(user);
we don't need do any extra thing since the field is static.

Reply:
Shouldn't we handle that logic in the main method, due to scope of the Curr object?